/*
 * Copyright Â© 2023- Gimlet Labs, Inc.
 * All Rights Reserved.
 *
 * NOTICE:  All information contained herein is, and remains
 * the property of Gimlet Labs, Inc. and its suppliers,
 * if any.  The intellectual and technical concepts contained
 * herein are proprietary to Gimlet Labs, Inc. and its suppliers and
 * may be covered by U.S. and Foreign Patents, patents in process,
 * and are protected by trade secret or copyright law. Dissemination
 * of this information or reproduction of this material is strictly
 * forbidden unless prior written permission is obtained from
 * Gimlet Labs, Inc.
 *
 * SPDX-License-Identifier: Proprietary
 */

syntax = "proto3";

package gml.gem.calculators.core.optionspb;

option go_package = "gimletlabs.ai/gimlet/src/gem/calculators/core/optionspb;optionspb";

import "google/protobuf/wrappers.proto";

message ByteTrackCalculatorOptions {
  // The number of frames a box is allowed to be lost before it is no longer tracked.
  google.protobuf.Int32Value max_frames_lost = 1;

  // Bytetrack thresholds (see https://github.com/ifzhang/ByteTrack).
  google.protobuf.FloatValue track_thresh = 10;
  google.protobuf.FloatValue high_thresh = 11;
  google.protobuf.FloatValue match_thresh = 12;
}
